github-actions/rebase-pr/entrypoint.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+set -e
+
+# Set email and name config
+git config --global user.email "devops+rebase@angular.io"
+git config --global user.name "Angular Rebase Bot"
+
+# Exit for code a successful run
+SUCCESS_EXIT_CODE=0
+
+# The repository name and branch name for the PRs head and base
+HEAD_REPOSITORY=$(jq -r ".pull_request.head.repo.full_name" "$GITHUB_EVENT_PATH")
+HEAD_BRANCH=$(jq -r ".pull_request.head.ref" "$GITHUB_EVENT_PATH")
+BASE_REPOSITORY=$(jq -r ".pull_request.base.repo.full_name" "$GITHUB_EVENT_PATH")
+BASE_BRANCH=$(jq -r ".pull_request.base.ref" "$GITHUB_EVENT_PATH")
+
+# Change to the github workspace and initialize the repo if needed.
+cd "$GITHUB_WORKSPACE"
+git init
```

Comment:
In what cases would the folder not be a Git repo?

Reply:
The git repo is only already available because in instance we have the `actions/checkout` action called before our `angular/dev-infra/github-actions/rebase-pr` action.

If we were to not have a step using `actions/checkout` beforehand, the repo would not be initialized yet.

Reply:
I see. would the `cd ""` then be incorrect if `GITHUB_WORKSPACE` is empty? Should we check for the `GITHUB_WORKSPACE` and otherwise create a tmp folder?

Reply:
It's provided by the system: https://help.github.com/en/articles/virtual-environments-for-github-actions#environment-variables

Reply:
The `GITHUB_WORKSPACE` directory is always created, but the `actions/checkout` action checks out the repo and places it there.

So if `GITHUB_WORKSPACE` is `/github/workspace/` then the repo would be there `/github/workspace/.git` being the location of the project git config (since that expresses where the repo is extracted at)